rust/lance-core/src/cache.rs

```diff
@@ -48,38 +49,104 @@ impl SizedRecord {
 /// The cache is keyed by the file path and the type of metadata.
 #[derive(Clone, Debug)]
 pub struct FileMetadataCache {
-    cache: Arc<Cache<(Path, TypeId), SizedRecord>>,
+    cache: Option<Arc<Cache<(Path, TypeId), SizedRecord>>>,
```

Comment:
There were (and perhaps still are) a lot of spots that use `Option<Cache>` with `None` to represent "no cache".  However, then you need special logic like...

```
let thing = if let Some(cache) {
  cache.find_or_insert(key, load_fn)
} else {
  load_fn()
}
```

Instead, I made it so you can cheaply create a `FileMetadataCache::no_cache()` which just never inserts anything.